tracing/trace_state.hh

```diff
@@ -239,7 +255,17 @@ private:
      */
     void begin() {
         std::atomic_signal_fence(std::memory_order_seq_cst);
-        _start = elapsed_clock::now();
+
+        auto steady_now = elapsed_clock::now();
+        if (_supplied_start_ts_us) {
+            // Set my start point to the POSIX TS from the primary session (but on `elapsed_clock`).
+            const elapsed_clock::time_point supplied_steady_start_point = elapsed_clock::now()
+                    + std::chrono::microseconds(*_supplied_start_ts_us)
+                    - std::chrono::system_clock::now().time_since_epoch();
+            _start = std::min(steady_now, supplied_steady_start_point);
+        } else {
+            _start = std::move(steady_now);
```

Comment:
Another alternative is to switch elapsed_clock to std::high_resolution_clock.

Reply:
On my chrono they point to the same type underneath. Do you mean it will look better? or be more future-proof?

Reply:
elapsed_clock is now steady_clock, no? If we switch to std::chrono::high_resolution_clock then we won't need to adjust for the different epochs.

Reply:
Ah, yes, but apparently not every impl of `high_resolution_clock` will work. Clang's would work: https://godbolt.org/z/Gr6WcdjeE
But cppreference says:
```
Class std::chrono::high_resolution_clock [...] It may be an alias of std::chrono::system_clock or std::chrono::steady_clock, or a third, independent clock.
```
So, if you don't mind, I'd better not touch it.

Reply:
> elapsed_clock is now steady_clock, no? If we switch to std::chrono::high_resolution_clock then we won't need to adjust for the different epochs.

Yes, but then we would break the invariant that `source_elapsed` is monotonically increasing.